apps/client/src/module/dashboard/profile/profile-service.ts

```diff
@@ -0,0 +1,65 @@
+"use server";
+
+import {
+  getServerGqlClient,
+  graphql,
+} from "@src/module/graphql";
+import { TypedDocumentNode } from '@graphql-typed-document-node/core';
+
+interface UpdateUserResult {
+  updateUser: boolean;
+}
+
+interface UpdateUserVariables {
+  id: string;
+  input: {
+    displayName: string;
+  };
+}
+
+const updateUserMutation = graphql(`
+    mutation UpdateUserDisplayName($id: String!, $input: UpdateUserInput!) {
+        updateUser(id: $id, input: $input)
+    }
+`) as TypedDocumentNode<UpdateUserResult, UpdateUserVariables>;
+
+export type UpdateUserPayload = {
+  id: string;
+  displayName: string;
+};
+
+function validateUpdatedDisplayName(name: string): boolean {
+    const regex = /^[a-zA-Z0-9-_]+$/;
+    return name.length >= 4 && name.length <= 15 && regex.test(name);
+  }
+
+export async function updateUserProfile(payload: UpdateUserPayload) {
+    if (!validateUpdatedDisplayName(payload.displayName)) {
+        return {
+          error: 'Display name must be 4-15 characters long and contain only letters, numbers, hyphens, and underscores.'
+        };
+    }
+
+    try {
+        const gqlClient = await getServerGqlClient();
+        const { updateUser } = await gqlClient.request(updateUserMutation, {
+          id: payload.id,
+          input: {
+            displayName: payload.displayName,
+          },
+        });
+    
+        if (updateUser) {
+          return { success: true };
+        } else {
+          return {
+            error: "Failed to update user profile",
+          };
+        }
+      } catch (error) {
+        console.error('Error updating user profile:', error);
```

Comment:
remove console log in PR unless it's for some essential purpose

Reply:
fixed